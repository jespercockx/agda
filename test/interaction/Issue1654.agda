{-# OPTIONS --rewriting #-}

data _==_ {A : Set} (a : A) : A â†’ Set where
  idp : a == a
{-# BUILTIN REWRITE _==_ #-}

postulate
  A : Set
  a b : A

r : a == b
r = {!idp!}
{-# REWRITE r #-}
