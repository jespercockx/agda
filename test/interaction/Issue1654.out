(agda2-status-action "")
(agda2-info-action "*Type-checking*" "" nil)
(agda2-highlight-clear)
(agda2-status-action "")
(agda2-info-action "*All Goals*" "?0 : a == b " nil)
((last . 1) . (agda2-goals-action '(0)))
(agda2-info-action "*Error*" "1,1-4 a != b of type A when checking that the expression idp has type a == b" nil)
(agda2-highlight-load-and-delete-action)
(agda2-status-action "")
